modules/regex/regex.cpp

```diff
@@ -326,6 +326,10 @@ String RegEx::sub(const String &p_subject, const String &p_replacement, bool p_a
 	pcre2_match_context_free_32(mctx);
 
 	if (res < 0) {
+		PCRE2_UCHAR32 buf[256];
+		pcre2_get_error_message_32(res, buf, 256);
+		String message = "PCRE2 Error: " + String((const char32_t *)buf);
+		ERR_PRINT(message.utf8());
 		return String();
```

Comment:
Perhaps in case of an error we should return the original string, and not an empty one?

Reply:
Good question. We could, but that should then also be changed for the early failure checks at the start of the function.

`sed` and Python `re` both raise exceptions so they don't have a fallback behavior to take inspiration from.

Interesting, `perl` doesn't error and instead returns the _replacement string_ with the unmatched group being an empty string:
```
$ perl -pe 's/((get_|set_)?)property/\5new_property/g' <<< "property get_property set_property"
new_property new_property new_property
```

That kind of makes sense?

Reply:
Also with a case that doesn't involve actual capture groups:
```
$ perl -pe 's/property/\5new_property/g' <<< "property get_property set_property"
new_property get_new_property set_new_property
```

If the pattern to replace is a match, and the replacement pattern can be made sense of by treating the invalid reference as empty, then I'd indeed expect it to do what perl does. (Though unlike perl, I'd still print an error that the replacement string references a non-existing capture group.)

Reply:
This makes sense to me logically, but since an unknown capture group throws an error anyway, I think we could just return an empty string. A fallback would require calling PCRE2 again with `PCRE2_SUBSTITUTE_UNKNOWN_UNSET`, so I'm not sure it's worth it.

Reply:
However, I tried to do it. Looks correct.